src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -262,7 +266,44 @@ private void initDragAndDrop() {
                             for (Tab libraryTab : tabbedPane.getTabs()) {
                                 if (libraryTab.getId().equals(tab.getId()) &&
                                         !tabbedPane.getSelectionModel().getSelectedItem().equals(libraryTab)) {
-                                    ((LibraryTab) libraryTab).dropEntry(stateManager.getLocalDragboard().getBibEntries());
+                                    LibraryTab libraryTabCasted = (LibraryTab) libraryTab;
+                                    if (dragboard.hasContent(DragAndDropDataFormats.GROUP)) {
+                                        List<String> groupPathToSources = (List<String>) dragboard.getContent(DragAndDropDataFormats.GROUP);
+                                        if (libraryTabCasted.getBibDatabaseContext().getMetaData().getGroups().isEmpty()) {
```

Comment:
Please consider reducing the level of nesting. Nested code is hard to follow. Physically there’s more visual scanning with the eyes. Mentally, each level of nesting requires more overhead to track functionality. All of this exhaust a reader.

A common solution to reduce nesting is by inverting if statements; You can read more about it [here](https://stackoverflow.com/questions/268132/invert-if-statement-to-reduce-nesting). In addition, you can replace for loops with Java 8 stream operations *(map-filter-flatmap)* when possible. It will make your code so much easier to follow.